pkg/backup/item_block_worker_pool.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright the Velero Contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backup
+
+import (
+	"context"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type ItemBlockWorkerPool struct {
+	inputChannel chan ItemBlockInput
+	wg           *sync.WaitGroup
+	logger       logrus.FieldLogger
+	cancelFunc   context.CancelFunc
+}
+
+type ItemBlockInput struct {
+	itemBlock  *BackupItemBlock
+	returnChan chan ItemBlockReturn
+}
+
+type ItemBlockReturn struct {
+	itemBlock *BackupItemBlock
+	resources []schema.GroupResource
+	err       error
+}
+
+func (p *ItemBlockWorkerPool) GetInputChannel() chan ItemBlockInput {
+	return p.inputChannel
+}
+
+func StartItemBlockWorkerPool(ctx context.Context, workers int, log logrus.FieldLogger) *ItemBlockWorkerPool {
+	// Buffer will hold up to 10 ItemBlocks waiting for processing
+	inputChannel := make(chan ItemBlockInput, 10)
```

Comment:
Are are going to make this configurable and when?
What is the consideration?

Reply:
See the above comment. I don't really see a big need to make it configurable -- the concern is that users might get confused with both a worker pool size configuration (how many worker threads, this config is already in place) and a channel buffer size configuration, which really won't affect performance much at all. I can certainly make this configurable, though, if there is a use case for it.

Reply:
Understood